### PR TITLE
feat(learn): Message Board User Stories

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>

Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Related #39721 

* Updates wording of test `text` to second voice
* Standardises description
* Migrates boilerplate instructions to /learn

### CODE REVIEW

File: curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md [hunk 1]
```diff
@@ -8,21 +8,31 @@ forumTopicId: 301568
...
+When you are done, make sure a working demo of your project is hosted somewhere public. Then submit the URL to it in the `Solution Link` field. Optionally, also submit a link to your projects source code in the `GitHub Link` field.
 </section>
 
 ## Instructions
 <section id='instructions'>
 
+1. Set `NODE_ENV` to test without quotes when ready to write tests and DB to your databases connection string (in `.env`)
+2. Recommended to create controllers/handlers and handle routing in `routes/api.js`
+3. You will add any security features to `server.js`
+4. You will create all of the functional/unit tests in `tests/2_functional-tests.js` and `tests/1_unit-tests.js` but only functional will be tested
```
Comment (RandellDawson) [inline curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md:26]: I am not understanding this line.  If unit tests are not going to be tested, then should we be telling the user to create tests in `/tests/1_unit-tests.js`?

File: curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md [hunk 2]
```diff
@@ -45,23 +55,23 @@ tests:
...
     testString: ''
-  - text: I can report a thread and change its `reported` value to `true` by sending a PUT request to `/api/threads/{board}` and pass along the `thread_id` (Text response will be 'success').
+  - text: You can send a PUT request to `/api/threads/{board}` and pass along the `thread_id`. Returned will be the string `success`. The `reported` value of the `thread_id` will be changed to `true`.
     testString: ''
-  - text: I can report a reply and change its reported value to true by sending a PUT request to `/api/replies/{board}` and pass along the `thread_id` & `reply_id` (Text response will be 'success').
+  - text: You can send a PUT request to `/api/replies/{board}` and pass along the `thread_id` & `reply_id`. Returned will be the string `success`. The `reported` value of the `reply_id` will be changed to `true`.
     testString: ''
-  - text: Complete functional tests that wholly test routes and pass.
+  - text: All 10 functional tests are complete and passing.
```
Comment (RandellDawson) [inline curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md:74]: When I look at the boilerplate, I can not tell how many tests there need to be.  There are not `test('test text', function(done) { })` as there are in other boilerplates.  Can the boilerplates be changes to explain what test to write like is done in PR https://github.com/freeCodeCamp/freeCodeCamp/pull/39976 ?

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@nhcarrigan See my comments/questions below?

File: curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md (hunk 1 above)
Comment (nhcarrigan) [reply curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md:26]: That was the original text - I'll remove the mention of the unit tests 🙂

File: curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md (hunk 2 above)
Comment (nhcarrigan) [reply curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md:74]: Sure thing - I will push an update later this afternoon 🙂

Comment (nhcarrigan) [reply curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md:74]: I've sent an update to the boilerplate PR for this.

File: curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md (hunk 1 above)
Comment (nhcarrigan) [reply curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md]: This has been reworded.

**Comment by nhcarrigan** [comment]
@RandellDawson I pushed a commit here that adds the test descriptions to the instructions - let me know your thoughts on this. Once we've ironed out the approach, I can mirror the changes on the Stock Price app.

/cc @moT01

File: curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md [hunk 3]
```diff
@@ -8,21 +8,43 @@ forumTopicId: 301568
...
-Working on this project will involve you writing your code on Repl.it on our starter project. After completing this project you can copy your public Repl.it URL (to the homepage of your app) into this screen to test it! Optionally you may choose to write your project on another platform but it must be publicly visible for our testing.
-Start this project on Repl.it using <a href="https://repl.it/github/freeCodeCamp/boilerplate-project-messageboard">this link</a> or clone <a href='https://github.com/freeCodeCamp/boilerplate-project-messageboard/'>this repository</a> on GitHub! If you use Repl.it, remember to save the link to your project somewhere safe!
+Working on this project will involve you writing your code using one of the following methods:
+
+- Clone <a href='https://github.com/freeCodeCamp/boilerplate-project-messageboard/'>this GitHub repo</a> and complete your project locally.
+- Use <a href='https://repl.it/github/freeCodeCamp/boilerplate-project-messageboard' target='_blank'>our repl.it starter project</a> to complete your project.
```
Comment (huyenltnguyen) [inline curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md:14]: The Repl link is opened in a new tab, while the GitHub one does not. I'm wondering if this is intentional

Comment (RandellDawson) [reply curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md:14]: @huyenltnguyen Good catch!  Would you be interested in updating all the backend project's with the `_blank` for the GitHub link in a separate PR?

Comment (nhcarrigan) [reply curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md:14]: I'll get this fixed on this PR. 🙂

Comment (RandellDawson) [reply curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md]: @huyenltnguyen Actually, I decided to go ahead and not only fix the other such project links, but also make sure they are using the new `description` we are using.  I should have a PR shortly.

**Review by moT01 — APPROVED** [review]
LGTM 🎉 

The only thing I could mention is that there was some concern about remove a table from the user stories on the demo-project (and maybe boilerplate) [described here](https://github.com/freeCodeCamp/demo-projects/pull/30) - I don't think we need to add it, the user stories describe it pretty good.

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
I have tested everything again and like these new changes.  I approve.

**Pull request merged**